src/Common/rename.cpp

```diff
@@ -0,0 +1,122 @@
+#include <Common/rename.h>
```

Comment:
I think file name is too general.
At least better to use `renameDirectory.h` or something.